### ISSUE TITLE
Trojan Alert on x32 download

### ISSUE THREAD

**Issue by MJAdamczyk** [NexarDeveloper/nexar-supply-excel-cs#5]
My network blocked the download for the x32 version of the plugin citing a Trojan threat.  Please review your build to ensure that this is fixed.

**Comment by MJAdamczyk** [comment]
Hi Rob!  Thank you for the quick response!  I suppose that could be the case.  I will reach out to IT and see if that's what prompted this and get back to you when I hear back from them.

I have attached an image below of the screen I am redirected to when I try to download the binary in case that helps you.

![image](https://user-images.githubusercontent.com/95430709/144463134-07dbb10b-fa56-4daa-8a2e-d2431812cc0b.png)

**Comment by joshcurrant** [comment]
Hi @RobAltium and @MJAdamczyk MJ

I've been using the 32 bit .xll, and just received the Trojan alert from Windows defender. I also uploaded it to virus total, and got the following result. Let me know if you can rebuild and see if that fixes the problem. 

![image](https://user-images.githubusercontent.com/96067971/145836873-26b30417-73b9-473f-b13f-629c05e32461.png)

Thanks!

**Comment by RobAltium** [comment]
> Hi @RobAltium and @MJAdamczyk MJ
> 
> I've been using the 32 bit .xll, and just received the Trojan alert from Windows defender. I also uploaded it to virus total, and got the following result. Let me know if you can rebuild and see if that fixes the problem.
> 
> ![image](https://user-images.githubusercontent.com/96067971/145836873-26b30417-73b9-473f-b13f-629c05e32461.png)
> 
> Thanks!
Thanks @joshcurrant

**Comment by RobAltium** [comment]
Hi @DrLock, @joshcurrant, @MJAdamczyk 

I did a little digging today.  It seems there is an issue with the ExcelDna library being wrongly flagged as a threat, including in Windows Defender after a recent update.  We are using this library to build the Nexar Excel add-in so see the false positive.

See, for example, this issue: https://github.com/Excel-DNA/ExcelDna/issues/403

It may be useful to report this to Microsoft via their inspection tool: https://www.microsoft.com/en-us/wdsi/filesubmission.

I see there was an update to the defender definitions, though we don't run that AV software here so I haven't been able to check the outcome: https://docs.microsoft.com/en-gb/microsoft-365/security/defender-endpoint/manage-updates-baselines-microsoft-defender-antivirus?view=o365-worldwide 

I'd love to know if that helps anyone.

**Comment by MJAdamczyk** [comment]
Hi Rob,

My company's firewall is blocking the download entirely, so I can't even check the Defender results.  I will go ahead and ask IT if they are using the most up-to-date definitions.  I'll keep you posted, but with Christmas coming next week, it may be next year before someone gets around to checking it.

**Comment by joshcurrant** [comment]
@RobAltium looks like the issue is getting resolved based on the latest posts https://github.com/Excel-DNA/ExcelDna/issues/403

Any chance you could rebuild and push a new version to possibly force antivirus to rescan the github download link? I am also blocked from downloading it. 
Thanks!

**Comment by RobAltium** [comment]
> @RobAltium looks like the issue is getting resolved based on the latest posts [Excel-DNA/ExcelDna#403](https://github.com/Excel-DNA/ExcelDna/issues/403)
> 
> Any chance you could rebuild and push a new version to possibly force antivirus to rescan the github download link? I am also blocked from downloading it. Thanks!

Hi @joshcurrant - that's hopeful news.  I have made a minor change to the add-in (functionality unchanged) and it is now at version 0.3.1.  The changes have been pushed and the links in this repo's README will point at the latest binaries.

I had mixed results with my tests.  Firstly, the 64-bit version downloaded fine straight away on my work's laptop with its bang-up-to-date AV.  On my home PC, which runs the 32-bit version, Windows AV still flagged the file as containing a virus.  Once I allowed this file it got through.  In the end I was able to test both 64-bit and 32-bity add-ins and confirm they work.

I think it may still depend on whether your organisation has the very latest AV definitions and your AV provider has updated.
